spec-tests/src/test/java/org/candlepin/spec/consumers/ConsumerFactsSpecTest.java

```diff
@@ -0,0 +1,195 @@
+import org.candlepin.spec.bootstrap.data.builder.Consumers;
+import org.candlepin.spec.bootstrap.data.builder.Facts;
+import org.candlepin.spec.bootstrap.data.builder.Owners;
+import org.candlepin.spec.bootstrap.data.util.UserUtil;
+
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.time.OffsetDateTime;
+
+@SpecTest
+public class ConsumerFactsSpecTest {
+
+    private static ApiClient admin;
+    private OwnerDTO owner;
+    private ApiClient ownerClient;
+
+    @BeforeAll
+    static void beforeAll() {
+        admin = ApiClients.admin();
+    }
+
+    @BeforeEach
+    void setUp() {
+        this.owner = admin.owners().createOwner(Owners.random());
+        this.ownerClient = ApiClients.basic(UserUtil.createAdminUser(admin, this.owner));
+    }
+
+    @Test
+    public void shouldAllowAddingFacts() {
+        ConsumerDTO consumer = createConsumer();
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        consumer.putFactsItem(Facts.MemoryTotal.key(), "100");
+        consumerClient.consumers().updateConsumer(consumer.getUuid(), consumer);
+
+        ConsumerDTO updatedConsumer = consumerClient.consumers().getConsumer(consumer.getUuid());
+
+        assertThat(updatedConsumer.getFacts())
+            .containsEntry(Facts.MemoryTotal.key(), "100");
+        assertThat(updatedConsumer.getInstalledProducts())
+            .map(ConsumerInstalledProductDTO::getProductName)
+            .containsExactly("Installed");
+    }
+
+    @Test
+    public void shouldAllowUpdatingFacts() {
+        ConsumerDTO consumer = createConsumer();
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        consumer.putFactsItem(Facts.UnameSystem.key(), "BSD");
+        consumerClient.consumers().updateConsumer(consumer.getUuid(), consumer);
+
+        ConsumerDTO updatedConsumer = consumerClient.consumers().getConsumer(consumer.getUuid());
+
+        assertThat(updatedConsumer.getFacts())
+            .containsEntry(Facts.UnameSystem.key(), "BSD");
+    }
+
+    @Test
+    public void shouldAllowRemovingFacts() {
+        ConsumerDTO consumer = createConsumer();
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+
+        consumer.getFacts().remove(Facts.UnameSystem.key());
+        consumerClient.consumers().updateConsumer(consumer.getUuid(), consumer);
+
+        ConsumerDTO updatedConsumer = consumerClient.consumers().getConsumer(consumer.getUuid());
+
+        assertThat(updatedConsumer.getFacts())
+            .doesNotContainKey(Facts.UnameSystem.key());
+    }
+
+    @Test
+    public void shouldUpdateConsumerUpdatedDateWhenFactsAreUpdated() {
+        ConsumerDTO consumer = createConsumer();
+        ApiClient consumerClient = ApiClients.ssl(consumer);
+        OffsetDateTime initialDate = consumer.getUpdated();
+
+        // MySQL drops millis, we need to wait a bit longer
+        sleepForMs(1000);
```

Comment:
Could probably just throw the InterruptedException from here instead of abstracting the whole thing out

Reply:
Done.